src/dataset.py

```diff
@@ -254,26 +254,29 @@ def get_activity_output_for_dataset(self, driver, uuid):
                 raise
 
     @classmethod
-    def get_dataset_access_level(self, uuid):
+    def get_entity_access_level(self, uuid):
         driver = None
         try:
-            query = "match (e:Entity {entitytype: 'Dataset'})-[:HAS_METADATA]-(m:Metadata) where e.uuid = '" + uuid + "' return m.data_access_level as acc_level"
+            query = "match (e:Entity)-[:HAS_METADATA]->(m:Metadata) where e.{uuid_attr} = '{uuid}' return m.{data_access_attr} as acc_level".format(
+                data_access_attr=HubmapConst.DATA_ACCESS_LEVEL,uuid_attr=HubmapConst.UUID_ATTRIBUTE,uuid=uuid)
             conn = Neo4jConnection(self.confdata['NEO4J_SERVER'], self.confdata['NEO4J_USERNAME'], self.confdata['NEO4J_PASSWORD'])
             driver = conn.get_driver()
             return_list = []
             with driver.session() as session:
-                    for record in session.run(query):
-                        #return_list.append(record)
+                for record in session.run(query):
+                    if record['acc_level'] is not None:
+                        print(record['acc_level'])
                         return_list.append(record['acc_level'])
             
+            print(return_list)
             if len(return_list) == 0:
-                raise HTTPException("Dataset uuid:" + uuid + " not found.", 404)
+                raise HTTPException("Entity uuid:" + uuid + " not found.", 404)
             if len(return_list) > 1:
-                raise HTTPException("Multiple datasets found for uuid:" + uuid, 500)
+                raise HTTPException("Multiple entities found for uuid: " + uuid, 500)
             return return_list[0]
                         
-        except Exception as e:
-            raise HTTPException("Unhandled Exception occurred", 500)
+        except CypherError as ce:
+            raise CypherError('A Cypher error was encountered: ' + ce.message)
```

Comment:
Same issue as above, we need to cleanly handle any exception that occurs, not just CypherErrors.  Unforseen, but uncommon, exceptions could occur in code (python or third-party libraries) that we don't control.